bidwire/scrapers/cityofboston_scraper.py

```diff
@@ -94,3 +106,61 @@ def scrape_bid_page(self, bid_id):
             items=items,
             site=self.get_site().name
         )
+
+
+class NoticesScraper(BaseScraper):
+    notices_url = "https://www.boston.gov/public-notices"
+
+    def __init__(self, threads=1, processes=4):
+        self.scraper = scrapelib.Scraper()
+        self.proc_executor = ProcessPoolExecutor(processes)
+        self.thread_executor = ThreadPoolExecutor(threads)
+
+    def scrape_desc(self, href):
+        desc_page = self.scraper.get('https://www.boston.gov' + href)
+        tree = html.fromstring(desc_page.content)
+        try:
+            return html.tostring(tree.xpath('//div["field-item even"=@class]')[0]).decode()
+        except:
+            # the description is a list
+            return html.tostring(tree.xpath('//div["body"=@class]/ol')[0]).decode()
+
+
+    def scrape_notice_div(self, div):
+        title_a = div.xpath(".//div['n-li-t'=@class]/a")[0]
+        year, month, day_and_start, end = div.xpath("//span['dc:date'=@property]/@content")[0].split('-')
+        day, start = day_and_start.split('T')
+        date = dtdate(int(year), int(month), int(day))
+        posted_candidates = div.xpath("//span['dl-d'=@class]")
+        for cand in posted_candidates:
+            if cand.text is None:
+                continue
+            if re.match('\d\d/\d\d/\d\d\d\d - \d:\d\d[ap]m', cand.text):
+                posted = datetime.strptime(cand.text, '%m/%d/%Y - %I:%M%p')
+                break
+        else:
+            raise ValueError("Couldn't get time of post")
+        return Document(
+            url='https;//www.boston.gov' + title_a.attrib['href'],
+            title=title_a.attrib['title'],
+            site=Document.Site.BOSTON.name,
+            description=self.scrape_desc(title_a.attrib['href'])
+        )
+
+    def get_site(self):
+        return Bid.Site.CITYOFBOSTON
```

Comment:
Thanks, I'll do all these

The proc_executor was meant to give a speedup but it wasn't necessary and caused its own problems.